source/virtualBuffers/__init__.py

```diff
@@ -366,7 +374,19 @@ def loadBuffer(self):
 
 	def _loadBuffer(self):
 		try:
-			self.VBufHandle=NVDAHelper.localLib.VBuf_createBuffer(self.rootNVDAObject.appModule.helperLocalBindingHandle,self.rootDocHandle,self.rootID,unicode(self.backendName))
+  			weblink=self.documentConstantIdentifier
+  			parsed_uri = urlparse( weblink )
+  		 	domain='{uri.netloc}'.format(uri=parsed_uri)
+   		 	domain=domain.replace('.','_')
+   		  	domain=domain.replace(':','_')
+   		  	domain=domain.replace('\\','_')
+   		  	filename=domain+'.ini'
+   		  	labels=""
+   		  	config = configobj.ConfigObj(os.path.join(globalVars.appArgs.configPath, "webLabels\%s" % filename))
+   		  	for k,v in config.iteritems():
+				labels+=k+":"+v+","
+			
+			self.VBufHandle=NVDAHelper.localLib.VBuf_createBuffer(self.rootNVDAObject.appModule.helperLocalBindingHandle,self.rootDocHandle,self.rootID,unicode(self.backendName),unicode(labels))

```

Comment:
As this is only supported for MSHTML so far, it might be best to add a class viarable on VirtualBuffer which can turn on and off support for labels. This would be set to True in mshtml's VirtualBuffer but False in the base VirtualBuffer, until all VirtualBuffers finally support labels. Otherwise you're reading in files and passing needless data when it willbe never used.